Fix wvdial so that it works with scandir as defined in uClibc.

Signed-off-by: Simon Dawson <spdawson@gmail.com>

diff -Nurp a/wvmodemscan.cc b/wvmodemscan.cc
--- a/wvmodemscan.cc	2009-09-29 18:27:28.000000000 +0100
+++ b/wvmodemscan.cc	2012-07-28 14:03:56.359729660 +0100
@@ -495,13 +495,13 @@ static int fileselect(const struct diren
 	// (no internal ISDN support)   || !strncmp(e->d_name, "ttyI", 4);
 }
 
-#if defined(__GLIBC__) && __GLIBC_PREREQ(2, 10)
+#if defined(__UCLIBC__) || (defined(__GLIBC__) && __GLIBC_PREREQ(2, 10))
 static int filesort(const dirent **e1, const dirent **e2)
 #else
 static int filesort(const void *_e1, const void *_e2)
 #endif
 {
-#if !(defined(__GLIBC__) && __GLIBC_PREREQ(2, 10))
+#if !(defined(__UCLIBC__) || (defined(__GLIBC__) && __GLIBC_PREREQ(2, 10)))
     dirent const * const *e1 = (dirent const * const *)_e1;
     dirent const * const *e2 = (dirent const * const *)_e2;
 #endif
